tiny_cnn/core/backend_tiny.h

```diff
@@ -213,18 +282,22 @@ class tiny_backend : public backend {
 
  private:
     /* Pointer to the convolution parameters */
-    conv_params* params_;
+    conv_params* params_c_;
+    deconv_params* params_d_;
     fully_params* params_f_;
 
     /* Pointer to the workers */
     std::vector<conv_layer_worker_specific_storage>* conv_layer_worker_storage_;
+    std::vector<deconv_layer_worker_specific_storage>* deconv_layer_worker_storage_;
     std::vector<max_pooling_layer_worker_specific_storage>* max_pooling_layer_worker_storage_;
     std::vector<std::vector<cnn_size_t>>* out2in_;
     std::vector<cnn_size_t>* in2out_;
 
     /* Pointers to parent class functions */
     std::function<void(const vec_t&, int)> copy_and_pad_input;
+    std::function<void(const vec_t&, int)> copy_and_unpad_output;
     std::function<void(const vec_t&, vec_t&)> copy_and_unpad_delta;
+    std::function<void(const vec_t&, vec_t&)> copy_and_pad_delta;
     std::function<void(const vec_t&, const vec_t&, vec_t&)> backward_activation;
 };
```

Comment:
@nyanp @Wangyida I think that we need to refactor the private attributes in a more generic way. Ideas?